tests/test_statelib_state.py

```
@@ -0,0 +1,112 @@
+
+import typing as tp
+
+import pytest
+
+import statelib as sl
+
+
+def test_flat_state_direct_init_raises() -> None:
+    with pytest.raises(
+        TypeError,
+        match="'FlatState' should never be directly instantiated, use 'FlatState.from_pytree' instead",
```

Ruff (0.14.0)

Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)